### PR TITLE
Azure HA Teleport deployment guide

### PR DESCRIPTION
**Pull request by espadolini**
This PR adds a HA deployment guide for Azure using the `teleport-cluster` chart, PostgreSQL for backend and audit log and Azure Blob Storage for session recordings.

### CODE REVIEW

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+```
+
+</Details>
+
+Now we'll configure the `teleport-cluster` Helm chart to use the `azure` mode. Create a
+file called `azure-values.yaml` containing the values you've selected above:
+
+<ScopedBlock scope={["oss"]}>
+
+```code
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:123]:
```suggestion
```yaml
```

(Assuming this renders OK with the code blocks, it's way more readable and breaks the page up)

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+chartMode: azure
+# Name of your cluster. Use the FQDN you intend to configure in DNS later
+clusterName: teleport.example.com
+azure:
+  databaseHost: <Var name="teleport-pg" />.postgres.database.azure.com
+  databaseUser: <Var name="pguser" />
+  sessionRecordingStorageAccount: <Var name="teleportblob" />.blob.core.windows.net
+  # Whether to mirror audit log entries to stdout in JSON format (useful for external log collectors)
+  auditLogMirrorOnStdout: false
+  clientID: $(az identity show --resource-group <Var name="teleport-rg" /> --name teleport-id --query clientId -o tsv)
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:134]: It's probably providing people with a way to get and then insert this themselves, as many people who read these guides don't understand what a heredoc is or how it works.

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+highAvailability:
+  # Number of replicas to configure
+  replicaCount: 2
+  certManager:
+    # Enable cert-manager support to get TLS certificates
+    enabled: true
+    # Name of the cert-manager GlobalIssuer or Issuer to use
+    issuerName: letsencrypt-production
+    issuerKind: GlobalIssuer
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:143]: Might be worth mentioning that you'll need to configure `cert-manager` indepedently.

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+```
+
+<Admonition type="note">
+  You cannot change the `clusterName` after the cluster is configured, so make sure you choose wisely. We recommend using the fully-qualified domain name that you'll use for external access to your Teleport cluster.
+</Admonition>
+
+Once the chart is installed, you can use `kubectl` commands to view the deployment:
+
+```code
+$ kubectl --namespace teleport get all
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:199]:
```suggestion
$ kubectl --namespace <Var name="teleport-ns" /> get all
```

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+deployment.apps/teleport-proxy   2/2     2            2           22h
+
+NAME                                       DESIRED   CURRENT   READY   AGE
+replicaset.apps/teleport-auth-57989d4cb    2         2         2       22h
+replicaset.apps/teleport-proxy-c6bf55cfc   2         2         2       22h
+```
+
+## Step 4/5. Set up DNS
+
+We'll now set up DNS `A` records for `teleport.example.com` and `*.teleport.example.com`.
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:224]: I'd add something about how this assumes that you're using a domain configured with Azure DNS, and to add the records in your own provider if you're not.

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+```code
+$ kubectl --namespace teleport exec deploy/teleport-auth -- tctl users add test --roles=access,editor
+
+User "test" has been created but requires a password. Share this URL with the user to complete user setup, link is valid for 1h:
+https://teleport.example.com:443/web/invite/91cfbd08bc89122275006e48b516cc68
+
+NOTE: Make sure teleport.example.com:443 points at a Teleport proxy that users can access.
+```
+</ScopedBlock>
+<ScopedBlock scope={["enterprise", "cloud"]}>
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:253]: This is a cluster deployment guide, so doesn't need the Cloud scope

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+```
+
+## Step 4/5. Set up DNS
+
+We'll now set up DNS `A` records for `teleport.example.com` and `*.teleport.example.com`.
+
+```code
+# Change these parameters if you altered them above
+$ NAMESPACE=teleport
+$ RELEASE_NAME=teleport
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:229]:
```suggestion
$ NAMESPACE=<Var name="teleport-ns" /> 
$ RELEASE_NAME=<Var name="teleport-release" />
```

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+```
+
+## Step 5/5. Create a Teleport user
+
+Create a user to be able to log into Teleport. This needs to be done on the Teleport auth server,
+so we can run the command using `kubectl`:
+
+<ScopedBlock scope={["oss"]}>
+```code
+$ kubectl --namespace teleport exec deploy/teleport-auth -- tctl users add test --roles=access,editor
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:245]:
```suggestion
$ kubectl --namespace <Var name="teleport-ns" /> exec deploy/<Var name="teleport-release" />-auth -- tctl users add test --roles=access,editor
```

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+
+User "test" has been created but requires a password. Share this URL with the user to complete user setup, link is valid for 1h:
+https://teleport.example.com:443/web/invite/91cfbd08bc89122275006e48b516cc68
+
+NOTE: Make sure teleport.example.com:443 points at a Teleport proxy that users can access.
+```
+</ScopedBlock>
+<ScopedBlock scope={["enterprise", "cloud"]}>
+```code
+$ kubectl --namespace teleport exec deploy/teleport-auth -- tctl users add test --roles=access,editor,reviewer
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:255]:
```suggestion
$ kubectl --namespace <Var name="teleport-ns" /> exec deploy/<Var name="teleport-release" />-auth -- tctl users add test --roles=access,editor,reviewer
```

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 10]
```diff
@@ -0,0 +1,325 @@
...
+
+<Tabs>
+  <TabItem label="Using values.yaml">
+  Edit your `azure-values.yaml` file from above and make the appropriate changes.
+
+  Upgrade the deployment with the values from your `azure-values.yaml` file using this command:
+
+  ```code
+  $ helm upgrade teleport teleport/teleport-cluster \
+    --namespace teleport \
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:290]:
```suggestion
  $ helm upgrade <Var name="teleport-release" /> teleport/teleport-cluster \
    --namespace <Var name="teleport-ns" /> \
```

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 11]
```diff
@@ -0,0 +1,325 @@
...
+    -f azure-values.yaml
+  ```
+
+  </TabItem>
+  <TabItem label="Using --set via CLI">
+  Run this command, editing your command line parameters as appropriate:
+
+  ```code
+  $ helm upgrade teleport teleport/teleport-cluster \
+    --namespace teleport \
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:300]:
```suggestion
  $ helm upgrade <Var name="teleport-release" /> teleport/teleport-cluster \
    --namespace <Var name="teleport-ns" /> \
```

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 12]
```diff
@@ -0,0 +1,325 @@
...
+  To change `chartMode`, `clusterName` or any `azure` settings, you must first uninstall the existing chart and then install
+  a new version with the appropriate values.
+</Admonition>
+
+## Uninstalling Teleport
+
+To uninstall the `teleport-cluster` chart, use `helm uninstall <release-name>`. For example:
+
+```code
+$ helm --namespace teleport uninstall teleport
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:316]:
```suggestion
$ helm --namespace <Var name="teleport-ns" /> uninstall <Var name="teleport-release" />
```

**Comment by webvictim** [comment]
Some of these `<Var />` replacements might be a bit over the top and not look great in comments, they're just nits.

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx (hunk 1 above)
Comment (espadolini) [reply docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:123]: `yaml` doesn't support `<Var>` tho :(

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx (hunk 3 above)
Comment (espadolini) [reply docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:143]: The "prerequisites" section has this already 😅
> You will also need an Azure DNS zone, and access to an AKS cluster with cert-manager
[installed](https://cert-manager.io/docs/installation/) and configured to [issue certificates for said Azure DNS zone](https://cert-manager.io/docs/configuration/acme/dns01/azuredns/).

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx (hunk 6 above)
Comment (espadolini) [reply docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:253]: In my defense, I copypasted this from one of the other guides LOL

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx (hunk 2 above)
Comment (espadolini) [reply docs/pages/deploy-a-cluster/helm-deployments/azure.mdx]: Done in 11af5ab

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx (hunk 5 above)
Comment (espadolini) [reply docs/pages/deploy-a-cluster/helm-deployments/azure.mdx]: 519d9fcda1

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx (hunk 6 above)
Comment (espadolini) [reply docs/pages/deploy-a-cluster/helm-deployments/azure.mdx]: Switched to `TabItem`s scoped for oss and enterprise.

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 13]
```diff
@@ -0,0 +1,335 @@
...
+  clientID: "<Var name="teleport-clie-ntid-uuid-123456789012" />"
+highAvailability:
+  # Number of replicas to configure
+  replicaCount: 2
+  certManager:
+    # Enable cert-manager support to get TLS certificates
+    enabled: true
+    # Name of the cert-manager GlobalIssuer or Issuer to use
+    issuerName: letsencrypt-production
+    issuerKind: GlobalIssuer
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:150]: I think this should be `ClusterIssuer` as per https://cert-manager.io/docs/concepts/issuer/?

```suggestion
    issuerKind: ClusterIssuer
```

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 14]
```diff
@@ -0,0 +1,335 @@
...
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:178]: Ditto above

```suggestion
    issuerKind: ClusterIssuer
```

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 15]
```diff
@@ -0,0 +1,335 @@
...
+value [using `helm upgrade` as detailed below](#upgrading-the-cluster-after-deployment).
+
+## Upgrading the cluster after deployment
+
+To make changes to your Teleport cluster after deployment, you can use `helm upgrade`.
+
+Helm defaults to using the latest version of the chart available in the repo, which will also correspond to the latest
+version of Teleport. You can make sure that the repo is up to date by running `helm repo update`.
+
+If you want to use a different version of Teleport, set the [`teleportVersionOverride`](../../reference/helm-reference/teleport-cluster.mdx#teleportversionoverride) value.
```
Comment (webvictim) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:288]: We should advise people to use `--version` instead to keep the deployed chart version in sync with the deployed Teleport version.

```suggestion
If you want to use a different version of Teleport, pass the `--version` argument to Helm:

```code
$ helm upgrade --version 14.0.0 \
  <Var name="teleport-release" /> teleport/teleport-cluster \
  --namespace <Var name="teleport-ns" /> \
  -f azure-values.yaml
```

**Review by webvictim — APPROVED** [review]

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx (hunk 13 above)
Comment (espadolini) [reply docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:150]: It should!

File: docs/pages/deploy-a-cluster/helm-deployments/azure.mdx [hunk 16]
```diff
@@ -0,0 +1,342 @@
...
+
+## Prerequisites
+
+(!docs/pages/kubernetes-access/helm/includes/teleport-cluster-prereqs.mdx!)
+
+In addition, you will need `azure-cli` 2.51 or later to follow along these
+instructions. Reference the Azure docs on [how to install the Azure
+CLI](https://learn.microsoft.com/cli/azure/install-azure-cli).
+
+After installing it, make sure you are logged in, by typing `az login`. This
```
Comment (zmb3) [inline docs/pages/deploy-a-cluster/helm-deployments/azure.mdx:28]:
```suggestion
After installing it, make sure you are logged in by typing `az login`. This
```

**Review by zmb3 — APPROVED** [review]

**Pull request merged**